src/app/campaign/store/selectors/campaign-flight.selectors.ts

```diff
@@ -147,24 +147,22 @@ export const selectCampaignFlightInventoryReportData = createSelector(
     const inventoryDays = flights.reduce(
       (days, flight) =>
         inventory[flight.id].days.reduce((acc, day) => {
-          const date = day.date.toISOString().slice(0, 10);
-          return {
-            ...acc,
-            [date]: {
-              ...acc[date],
-              [flight.id]: day
-            }
+          const date = day.date.valueOf();
+          acc[date] = {
+            ...acc[date],
+            [flight.id]: day
           };
+          return acc;
         }, days),
-      {}
+      {} as { [date: number]: { [flightId: number]: FlightDay } }
```

Comment:
This code can be hard to read and understand. Sorry I didn't add this typing in here to begin with. It does seem to help to know what the heck it's trying to build here.